src/mod/applications/mod_callcenter/mod_callcenter.c

```diff
@@ -1168,7 +1205,27 @@ cc_status_t cc_agent_update(const char *key, const char *value, const char *agen
 			}
 			switch_safe_free(sql);
 		}
+	} else if (!strcasecmp(key, "not_registered_delay_time")) {
+		sql = switch_mprintf("UPDATE agents SET not_registered_delay_time = '%ld', instance_id = 'single_box' WHERE name = '%q'", atol(value), agent);
+		cc_execute_sql(NULL, sql, NULL);
+		switch_safe_free(sql);
+		result = CC_STATUS_SUCCESS;
+	} else if (!strcasecmp(key, "max_rejected")) {
+		sql = switch_mprintf("UPDATE agents SET max_rejected = '%d', instance_id = 'single_box' WHERE name = '%q'", atoi(value), agent);
+		cc_execute_sql(NULL, sql, NULL);
+		switch_safe_free(sql);
+		result = CC_STATUS_SUCCESS;
+	} else if (!strcasecmp(key, "max_busy")) {
+		sql = switch_mprintf("UPDATE agents SET max_busy = '%d', instance_id = 'single_box' WHERE name = '%q'", atoi(value), agent);
+		cc_execute_sql(NULL, sql, NULL);
+		switch_safe_free(sql);
+		result = CC_STATUS_SUCCESS;
 
+	} else if (!strcasecmp(key, "max_not_registered")) {
+		sql = switch_mprintf("UPDATE agents SET max_not_registered = '%d', instance_id = 'single_box' WHERE name = '%q'", atoi(value), agent);
+		cc_execute_sql(NULL, sql, NULL);
+		switch_safe_free(sql);
+		result = CC_STATUS_SUCCESS;
```

Comment:
Not required... But all these simple update which have the SQL column have the same name of the parameters could be groupped within the same else if.  I should have done this in the first place when I made it ! 

Also, I haven't fully looked up the code but does these change generate an event of this change ? If it doesn't, then this should be added (Again, not required per say, since it doesn't do it for the other one, but since you are in that code... :p   I could make a pull request inside your repo as well if you want me to do it.

Reply:
i think i understand what you mean, i will do the change if is not ok,  feel free to change it on my branch. 

no, there is no event generated when changing this parameters, i guess the events should be generated for all settings inside this function (ready_time, wrap_up_time, type, ..) ?
should i do this change now or leave it for another pr?

Reply:
the idea of replacing the else if, we probably could use a switch/case, to be able to have a default: result = CC_STATUS_INVALID_KEY;

what do you think?

Reply:
Yes and no...  To make use of a switch, we would have to generate ENUM for these options (like we did for the agent,  status and states..   In C, you can't do a switch on a string.
I'm not sure if it would bring much values to do so... It could be used to do a loop of all the argument and auto generate the api list.  But I think just using else if is fine.  You just need to capture all of the string we want in the same if condition and then in that if statement change the sql query with the provided parameters that was provided.